### PR TITLE
db: disable DQS on Python >= 3.12

### PR DESCRIPTION
**Pull request by wisp3rwind**
cf. https://github.com/beetbox/beets/issues/4709, let's see how badly this breaks CI

### CODE REVIEW

**Comment by wisp3rwind** [comment]
ah, the Ubuntu LTS on CI has an ancient SQLite...

**Comment by sungo** [comment]
to preface, I know precisely jack about python. I'm cutting and pasting to try and get my setup back online :) but after rolling a python 3.12 pyenv on freebsd, I tried this patch. it doesn't work for me unless those 0s are changed to 1s. so
`conn.setconfig(sqlite3.SQLITE_DBCONFIG_DQS_DDL, 1)` etc. 

with that change, I seem to be up and running again on freebsd 13.1

**Comment by wisp3rwind** [comment]
Oh, but if DQS can still be enabled at runtime, we could merge that change quickly as a workaround until we really remove all DQS usage (at least if the latter turns out to be a lot of work).

**Comment by wisp3rwind** [comment]
Oh, but if DQS can still be enabled at runtime, we could merge that change quickly as a workaround until we really remove all DQS usage (at least if the latter turns out to be a lot of work).

**Comment by wisp3rwind** [comment]
Rebased on master to see what happens with #5447, but CI cannot properly test this these days: https://github.com/beetbox/beets/issues/5450

File: beets/dbcore/db.py [hunk 1]
```diff
@@ -1070,6 +1071,16 @@ def _create_connection(self) -> Connection:
             # call conn.close() in _close()
             check_same_thread=False,
         )
+
+        if sys.version_info >= (3, 12) and sqlite3.sqlite_version_info >= (
+            3,
+            29,
```
Comment (snejus) [inline beets/dbcore/db.py:1130]: Why is the version `3.29.0` instead of `3.41.0` as you specified in the [issue description](https://github.com/beetbox/beets/issues/4709)?

> This is due to Double-quoted String Literals being disabled by default in the 3.41.0,1 version of SQLite

Comment (wisp3rwind) [reply beets/dbcore/db.py:1130]: 3.29 added the the config options used here, 3.41 changed the defaults; see https://www.sqlite.org/changes.html

So 3.29 is the earliest version for which we can ensure to disable DQS. Starting from 3.41 it's disabled anyway with default build options, but it seems prudent to me to enforce it here to ensure that we won't regress.

Comment (snejus) [reply beets/dbcore/db.py:1130]: I see, thanks. What would you think about removing this conditional logic and instead *always* enforcing double quotes?

Comment (wisp3rwind) [reply beets/dbcore/db.py:1130]: Now, I'm confused as to what you propose: double quoted string literals are the problematic feature, we want to _prohibit_ them so that we're sure to immediately see an error if a new instance of them is added accidentally. (Rather than only get a bug report later on when someone runs beets with a newer SQLite, which happened in #4709.) The idea with this PR was:

- SQLite < 3.29: DQS enabled by default, but nothing we can do about it, since SQLite doesn't have those config options
- SQLite >= 3.29, < 3.41: DQS enabled by default, but we can disable them
- SQLite >= 3.41, < 3.41: DQS disabled by default, no harm in disabling them; in fact, we might want to since people might build SQLite with different defaults (not sure whether anybody does)

We also need Python 3.12 to do anything, because that added `setconfig`.

In other words, if we attempt run this unconditionally, I'd expect either the Python library or SQLite itself to error out.

The relevance of this PR is probably decreasing, since newer version of Python and SQLite will become more prevalent. I'd argue to add it anyway, it's pretty simple, and it makes obvious and certain that DQS are disabled.

Comment (snejus) [reply beets/dbcore/db.py:1130]: Apologies, I was meant to say the opposite: always enforcing **single** quotes 🤦🏼

Comment (wisp3rwind) [reply beets/dbcore/db.py:1130]: Ok :+1: But, as I tried to explain, I don't think there's a way to do that unconditionally, at least not via the mechanism used here. Of course, the idea would be that DQS are generally disallowed in beets. We can't enforce that in every environment where beets might be run. This PR tries to make it more certain that we actually verify this at least in some cases (e.g. some CI runs): The problem with https://github.com/beetbox/beets/issues/4709 was that none of our CI runs actually used a version of SQLite where DQS are disabled by default, so we never saw the issue. Maybe the best way to think of this PR is indeed as an additional CI check.

**Comment by snejus** [comment]
I should have approved this a while ago, apologies @wisp3rwind.
